### ISSUE TITLE
Allow variable numbers of vertices in asteroids

### ISSUE THREAD

**Issue by mcdenhoed** [johnthagen/rust-belt#124]
Right now, our asteroids have a hard-coded-at-compile-time number of vertices:

```rust
/// The number of segments in an asteroid's shape is currently set at compile-time.
const NUM_SEGMENTS: usize = 20;
...
/// `CircularPolygon` is a piece of syntactic sugar to avoid having to use that ugly list type.
type CircularPolygon = [[f64; 2]; NUM_SEGMENTS];
```

Later on in the draw call, we use the `Asteroid.shape` attribute (of type `CircularPolyton`) in the draw calls  as follows:
```rust
        // This polygon is the "main" asteroid shape within the frame. It is
        // drawn at the location specified in `pos`. The CircularPolygon type,
        // being a list of lists of length 2, is an acceptable "shape" for
        // the polygon function
        polygon(color::WHITE,
                &self.shape,
                context
                    .transform
                    .trans(self.pos.x, self.pos.y)
                    .rot_rad(self.rot),
                graphics);
```

When we start splitting up asteroids, we will probably not want to be forced to have all of them to have the same number of vertices. Since `polygon` is already taking a reference to an array, we should be able to pass it a slice instead of a normal reference.

I propose that we add an attribute with a name like `num_vertices` to `Asteroid`, and use it as an ending index for a slice, and pass that instead of a normal reference. Will also need to refactor the constructor and asteroid-generation to take a number of vertices.

Doing this will help make asteroid splitting a thing.

@johnthagen We had discussed pros and cons of doing this with a vector vs a slice. Not sure about converting from a vector to an array ref for the draw call, or how performant it would be? We could store in a vector during generation and then "bake" it into an array afterwards?

**Comment by johnthagen** [comment]
@mcdenhoed I don't think the performance if using a `Vec` will be an issue at all, it's heavily optimized and it seems like if we need a variable length array that would be the first thing we should try. What I really like about `Vec` is that the length is stored explicitly inside the `Vec` rather than having to do some kind of implicit relationship outside with some kind of "array and length" type of thing you'd have to do in C.

I hope `Vec` has an easy `.as_ref()` or similar to make this easy.

But I think we'll know more after trying one of the solutions out, that's just my first reaction.

**Comment by johnthagen** [comment]
Subtask of #63.

**Comment by mcdenhoed** [comment]
See #125 for implementation

**Comment by mcdenhoed** [comment]
Closed in #125
